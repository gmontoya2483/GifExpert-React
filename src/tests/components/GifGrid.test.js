import {shallow} from "enzyme";
import AddCategory from "../../components/AddCategory";
import GifGrid from "../../components/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en componente <GifGrid/>', ()=> {
    const category = 'WTF'


    test('debe de mostrarse correctamnte', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes con nuestro Hook useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
            {
                id: 'DEF',
                url: 'https://localhost/cualquier/cosa.jpg',
                title: 'Cualquier cosa'
            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBeGreaterThanOrEqual(1)
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length)




    });




});
