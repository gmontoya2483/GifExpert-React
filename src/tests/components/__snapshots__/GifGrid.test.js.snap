// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid/> debe de mostrar items cuando se cargan imagenes con nuestro Hook useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    WTF
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente <GifGrid/> debe de mostrarse correctamnte 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    WTF
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading ... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
